### DISCUSSION TITLE
What are the best settings to preserve the quality and color of an original image?

### DISCUSSION THREAD

**Discussion by wes-kay** [ImageMagick/ImageMagick#5263] [Help]
I read this blog (great content) https://www.smashingmagazine.com/2015/06/efficient-image-resizing-with-imagemagick/ but I'm looking for the best quality not the most optimized.

Current command: `magick mogrify -unsharp 0.25x0.25+8+0.065 -resize 256x256\! -density 180 *.png`

**Comment by fmw42** [comment]
Type of resize filter to be used depends upon image content and desired mitigation of different types of artifacts. I do not agree with doing unsharp before the resize. I do not think that works well. 

See

https://imagemagick.org/Usage/filter/
https://imagemagick.org/Usage/filter/nicolas/

**Reply by wes-kay** [reply]
I don't fully understand what you mean as your posted examples do a resize after filters: 
![image](https://user-images.githubusercontent.com/40011470/175200380-747af40b-982a-497c-a3ef-bc3ff17b8e1b.png)

What command would you use then?

**Reply by fmw42** [reply]
Yes, you have to specify the -filter setting before -resize to have it be active, esp. in IM 7.  But you are doing -unsharp before -resize and in my opinion you should do -unsharp after -resize and use -unsharp 0xS+... for some value of sigma S.  Specifying both radius and sigma can often not be very good if you do not understand what they do. Best to specify only the Sigma with radius=0 so that IM will figure out the best radius to use with that Sigma.

**Reply by wes-kay** [reply]
Could you hit me with the sigma flags as I can't seem to find the correct command on the api.

current command: `magick mogrify -filter LanczosSharp -resize 256x256! -unsharp 0.25x0.25+8+0.065 -density 180 *.png`

**Reply by fmw42** [reply]
A rule of thumb is radius x sigma = 0xSigma will make a radius that is about 3xsigma.  See https://imagemagick.org/Usage/blur/. From Anthony on that page:

> The ideal solution is to simply set radius to '0x' as shown by the last line of the above table. In that case the operator will try to automatically determine the best radius for the sigma given. The smallest radius IM would use is 3, and is typically 3 * sigma for a Q16 version of IM (a smaller radius is used for IM Q8, as it has less precision). The only time I would use a non-zero radius was for a very small sigma or for specialized blurs. So..

> When possible use a radius of zero for blurring operations

The same holds for sharpening and unsharp.

**Comment by snibgo** [comment]
I agree with fmw42 about resizing first, then unsharp.

fmw42 also linked to a page by Nicolas Robidoux. When we want to downsize for the web, increasing sharpness without also adding halos, Nicolas's methods are very good. Much better quality than resize and unsharp, though the commands are more complex. I compare the methods at [Resampling with halo minimization](http://im.snibgo.com/resamphm.htm).
